### PR TITLE
Set C++ version to 17 if using Gadgetron (#1130)

### LINKED ISSUE #1129: Cannot set compiler flag -std=gnu++14 for g++-8 on VM
Trying to build SIRF on one of my VMs, I get lots of build errors like
```
‘enable_if_t’ in namespace ‘std’ does not name a template type
```
which, according to StackOverflow, can be cured by the compiler flag `-std=c++14` or `-std=gnu++14`.

I did try to add `-std=gnu++14` to `CMAKE_CXX_FLAGS` in `ccmake`, but it did not work: the actual compilation line showed by `make VERBOSE=1` turned out to be
```
cd /home/sirfuser/devel/build/builds/SIRF/build/src/xGadgetron/cGadgetron && /usr/bin/g++-8 -DBOOST_ALL_NO_LIB -DGADGETRON_TOOLBOXES_AVAILABLE -D_WIN32_WINNT=0x0501 -I/home/sirfuser/devel/build/sources/SIRF/src/xGadgetron/cGadgetron -I/home/sirfuser/devel/build/sources/SIRF/src/xGadgetron/cGadgetron/include -I/home/sirfuser/devel/build/sources/SIRF/src/iUtilities -I/home/sirfuser/devel/build/sources/SIRF/src/iUtilities/include -I/home/sirfuser/devel/build/sources/SIRF/src/common -I/home/sirfuser/devel/build/sources/SIRF/src/common/include -I/home/sirfuser/devel/build/sources/SIRF/src/Synergistic/cSyn -I/home/sirfuser/devel/build/sources/SIRF/src/Synergistic/cSyn/include -I/home/sirfuser/devel/build/sources/SIRF/src/xSTIR/cSTIR/include -I/home/sirfuser/devel/build/sources/SIRF/src/xSTIR/cSTIR -I/home/sirfuser/devel/build/sources/SIRF/src/Registration/cReg/include -I/home/sirfuser/devel/build/sources/SIRF/src/Registration/NiftyMoMo/include -isystem /home/sirfuser/devel/build/INSTALL/include -isystem /usr/include/hdf5/serial -isystem /home/sirfuser/devel/build/INSTALL/include/ITK-4.13 -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_SYSTEM_NO_DEPRECATED -std=gnu++14 -O3 -DNDEBUG -std=gnu++14 -fPIC -fopenmp -std=gnu++11 -MD -MT src/xGadgetron/cGadgetron/CMakeFiles/cgadgetron.dir/cgadgetron.cpp.o -MF CMakeFiles/cgadgetron.dir/cgadgetron.cpp.o.d -o CMakeFiles/cgadgetron.dir/cgadgetron.cpp.o -c /home/sirfuser/devel/build/sources/SIRF/src/xGadgetron/cGadgetron/cgadgetron.cpp
```
resulting expectedly in the same error
```
In file included from /home/sirfuser/devel/build/INSTALL/include/gadgetron/hoNDArray.h:7,
                 from /home/sirfuser/devel/build/sources/SIRF/src/xGadgetron/cGadgetron/include/sirf/Gadgetron/NonCartesianEncoding.h:68,
                 from /home/sirfuser/devel/build/sources/SIRF/src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_x.h:53,
                 from /home/sirfuser/devel/build/sources/SIRF/src/xGadgetron/cGadgetron/cgadgetron.cpp:38:
/home/sirfuser/devel/build/INSTALL/include/gadgetron/NDArray.h:60:14: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type
```
Any idea why/how/when `-std=gnu++11` got added to `CMAKE_CXX_FLAGS` and how to get rid of it?

### PR DESCRIPTION
**Pull request by evgueni-ovtchinnikov**
## Related issues

Fixes #1129.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added docstrings/doxygen in line with the guidance in the developer guide
- [ ] I have implemented unit tests that cover any new or modified functionality
- [x] The code builds and runs on my machine
- [ ] CHANGES.md has been updated with any functionality change

## Contribution Notes

Please read and adhere to the [contribution guidelines](https://github.com/SyneRBI/SIRF/blob/master/CONTRIBUTING.md).

Please tick the following: 

 - [x] The content of this Pull Request (the Contribution) is intentionally submitted for inclusion in SIRF (the Work) under the terms and conditions of the [Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0) License.

### CODE REVIEW

File: cmake/SetC++Version.cmake [hunk 1]
```diff
@@ -19,11 +19,15 @@
 # A macro to set the C++ version
 # If CMAKE_CXX_STANDARD is already set to a more recent version, keep that.
 macro(UseCXX VERSION)
-  if (NOT DEFINED CMAKE_CXX_STANDARD)
-    set (CMAKE_CXX_STANDARD ${VERSION})
-  else()
-    if ((CMAKE_CXX_STANDARD EQUAL 98) OR (CMAKE_CXX_STANDARD LESS VERSION))
-      message(FATAL_ERROR "CXX_STANDARD needs to be at least ${VERSION}")
+  if (CMAKE_VERSION VERSION_LESS "3.1")
```
Comment (KrisThielemans) [inline cmake/SetC++Version.cmake:22]: we don't need this as we require a more recent CMake

**Review by KrisThielemans — CHANGES_REQUESTED** [review]

**Review by KrisThielemans — COMMENTED** [review]
ok great. let's see if GHA is happy.

Maybe you could revert the white-space changes in `SetC++Version.cmake` then (with a `[ci skip]`), i.e. make it identical to the version `master`.

I'd then squash-merge it, but please edit the default message (which contains all commit messages) to something sensible (also making sure there's no `[ci skip]' in the squash-merge commit message).

**Comment by evgueni-ovtchinnikov** [comment]
> did it resolve your Gadgetron master problem?

using g++9 for building SIRF did

**Review by KrisThielemans — APPROVED** [review]

**Comment by KrisThielemans** [comment]
want me to squash-merge?

**Comment by evgueni-ovtchinnikov** [comment]
@KrisThielemans yes please (I am not sure I understand what you wrote about default message)

**Pull request merged**